### ISSUE TITLE
[J4.2.5] - Improve Joomla 4 Postinstall Messages

### ISSUE THREAD

**Issue by coolcat-creations** [joomla/joomla-cms#39362]
### Steps to reproduce the issue
If you make an update from a previous Joomla Version before 4.2.x, you get this After Install Messages

![grafik](https://user-images.githubusercontent.com/828371/205890575-b07045fc-d73d-45f3-b629-793d291060f1.png)

If you make a fresh install Joomla of 4.2.5 you only get Post Install Messages from Joomla 3
![grafik](https://user-images.githubusercontent.com/828371/205890784-7d23a257-acc0-4129-bdc2-42968a5f85bf.png)

From the following discussion, I understood that News about the new Plugins like HTTP headers or webauth disappear when the plugins get enabled. So in a fresh installation of 4.2.5 you won't get informed about new features in Joomla, but you get informed about two new resources that have been invented 9 Years ago. 

It does not make sense for me to highlight a change that has been made 9 Years ago but not mention a change that has been made for the recent version. 

Post Install messages are way too prominent to point out a fact that was invented 9 Years ago ?

**Comment by richard67** [comment]
@coolcat-creations Is there something missing or a typo in your description. For the first case you wrote "Install a 4.2 Joomla Version", and for the second case you wrote "Install a 4.2.5 Joomla Version". What's the difference? Should the first one be "Install a 4.2 Joomla Version 4.2.4 or older" or something like that? Or is the 2nd case not a new install but an update?

**Comment by coolcat-creations** [comment]
Its both separate installs, I changed the text to make it more clear

**Comment by richard67** [comment]
So the first "4.2" should have been a "4.2.0". That is not "more clear", it is just "clear", because before that change it was completely unclear.

**Comment by coolcat-creations** [comment]
> So the first "4.2" should have been a "4.2.0". That is not "more clear", it is just "clear", because before that change it was completely unclear.

yes sorry, for me a 4.2 without more information is a 4.2.0 but now hopefully its understandable

**Comment by richard67** [comment]
@coolcat-creations The postinstall message for improved multifactor authentication is only shown when all plugins of type "multifactorauth" are disabled, which is the case after an update to 4.2.0 but not after a new installation.

As soon as you disable all these plugins, that postinstall message will show up again. Could you check if this is the case on your 4.2.5 installation?

And can you check on the 4.2.0 if at least one of these plugins is enabled and if so, disable it to see if the message disappears?

If all that is the case, I would say for this one postinstall message it's expected behaviour.

**Comment by richard67** [comment]
So either your 4.2.0 was not a clean new install but an update, or you had disabled the named plugins after the installation.

**Comment by richard67** [comment]
The messages for Joomla 3 which you've mentioned are also shown on a clean, new installation of the current 4.2-dev branch. So all in all I see no issue here if the status of these plugins is like I wrote.

**Comment by coolcat-creations** [comment]
So Joomla 4 shows only Joomla 3 After Install messages and that's indended?

**Comment by richard67** [comment]
The statistics one is still relevant, so it is not Joomla 3 only, and the welcome thing I don't know if that still makes sense. Have you verified the statuses of your plugins like I wrote? Or did you not read what I wrote about that? Please report back if they are enabled or disabled so we can know if it is right or wrong that the messages about HTTP headers and improved multifactor auth are not shown.

**Comment by richard67** [comment]
P.S.: The version numbers are clearly labelled with a "Since version". So what makes you think that messages which exist since a version 3.x are only to be shown on Joomla 3 and not on Joomla 4? The statistics message is still relevant, as I said, and the hyperlinks in the welcome message are still working and still relevant, too. So your expectation that these are messages for Joomla 3 is simply wrong.

**Comment by coolcat-creations** [comment]
Hi Richard, I read what you wrote, but I do not understand it.
Messages are shown that Community resources are available for new users since Joomla 3.2 - when was that ? 15 years ago? Is this still worth to mention as a after install message ? Joomla 4 is a new major version so I think we should leave changes from J3 behind as J4.0.0 is the new "base"
Second: When I disable Http Headers the after install message for it appears in the post install messages. That makes no sense. Is it a post install message to inform about changes since a certain version? Then it should be shown independent of the status of this plugin. When I disable WebAuth no message appears. so either this is wrong or the message about http headers is wrong. The messages should be consistent no matter if there is an update or new installation.

**Comment by richard67** [comment]
>  When I disable WebAuth no message appears.

Where? In the user settings? These are just for an individual user. I was talking about the status of all plugins belonging to plugin group "webauthentication".

These 2 messages shall remind the user to switch on certain plugins. If he has done that or it was done with a new installation, there is no need to show them.

In general, when we update installations, we always add the new plugins with disabled status in our update SQL scripts because we don't know in all cases if the conditions to enable that particular plugin are fulfilled. This is why on a new installation such new plugins are enabled, but after an update from a previous version where they did niot exists so they were added, they are not enabled.

I am not the one who has invented these postinstall messages.

I am also not the one who has invented how we handle inserting of new plugins in update SQL scripts.

I am only the one who understood how it was meant to work and who tries to explain that here.

You might modify title and description of this issue or make new issues if you think something is wrong.

But like it is now this issue is not right because it describes a bug where there is no bug.

**Comment by brianteeman** [comment]
Would removing the "since version" part of the message be better?

**Comment by richard67** [comment]
> Would removing the "since version" part of the message be better?

@brianteeman You mean remove only the label, „Since version“ but leave the version? Or remove it all, including the version? The latter could be good, less confusing. If a particular postinstall message needs to refer to a version number, it can do that inside the message, e.g. „Beginning with version x, there is bla bla blubb“.

**Comment by coolcat-creations** [comment]
I edited the initial post completely

**Comment by brianteeman** [comment]
This should be looked at again for j6
